### PR TITLE
Hono examples

### PR DESCRIPTION
**Pull request by ToastedDev**
adds `github-oauth` and `username-and-password` examples for [hono](https://hono.dev).

### CODE REVIEW

**Review by pilcrowOnPaper — COMMENTED** [review]
Can you also run `pnpm format` in the repo root?

File: hono/github-oauth/routes/login/github.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { OAuth2RequestError, generateState } from "arctic";
+import { github, lucia } from "../../lib/auth.js";
+import { parseCookies, serializeCookie } from "oslo/cookie";
```
[the remark this answers is not in the document]
Comment (ToastedDev) [reply hono/github-oauth/routes/login/github.ts:3]: I just ported the Express examples into Hono. though you are right, we should use `hono/cookie` instead

File: hono/github-oauth/routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+mainRouter.get("/", async (c) => {
+	const user = c.get("user");
+	if (!user) {
+		return c.redirect("/login");
+	}
+	const templateFile = await fs.readFile("routes/index.template.html");
+	let template = templateFile.toString("utf-8");
+	template = template.replaceAll("%username%", user.username);
+	template = template.replaceAll("%user_id%", user.id);
+	return new Response(template, {
```
Comment (pilcrowOnPaper) [inline hono/github-oauth/routes/index.ts:17]: I believe you can't use `new Response()` if you want the `c.header()` in the middleware to work

**Review by pilcrowOnPaper — APPROVED** [review]
Thank you for the PR! Super sorry for the late review!

**Pull request merged**